src/PDBSetup.cpp

```diff
@@ -155,6 +155,49 @@ void Atoms::Clear()
   box.clear();
   resNames.clear();
   count = 0;
+  for (uint b = 0; b < BOX_TOTAL; b++) {
+    numAtomsInBox[b] = 0;
+  }
+}
+
+// This method of finding minimum assumes all the box 0 atoms 
+// will be contiguous in the coordinates array.  This isn't 
+// the case on checkpoint restarts.  Since we went out of the
+// way to ensure even when box transfers occur, the atoms
+// remain in the same original order in the original data structure.
+// We therefore can't rely on the molecule lookup to get the start 
+// and end of the box for the restarted data structures.
+// Hence the numberOfAtoms array.
+
+void Atoms::GetMinMaxAtoms(const uint b){
+  int stRange, endRange;
+  // -1 because we want to exclude the last array index  
+  // Box 0
+  // [0, numAtomsBox0)
+  // Box 1
+  // [numAtomsBox0, numAtomsBox0 + numAtomsBox1)
+  
+  // To prevent segfault
+  if (numAtomsInBox[b] == 0)
+    return;
+
+  if (b == 0){
+    stRange = 0;
+    endRange = numAtomsInBox[0] - 1;
+  } else if (b == 1) {
+    stRange = numAtomsInBox[0];
+    endRange = stRange + numAtomsInBox[1] - 1;
+  } else {
+    std::cout << "Error: Only Box 0 and Box 1 supported!" << std::endl;
+    exit(EXIT_FAILURE);
+  }
```

Comment:
Can't you make it more general to support simulation with more than 2 boxes?

Reply:
Calculate an atom offset array.  Now we just use start = offset[b], end = offset[b+1]-1.  This is general to any amount of box numbers.